src/Common/src/Interop/Windows/BCrypt/RC2BCryptModes.cs

```diff
@@ -10,29 +10,33 @@ namespace Internal.Cryptography
 {
     internal static class RC2BCryptModes
     {
-        private static readonly SafeAlgorithmHandle s_hAlgCbc = OpenRC2Algorithm(Cng.BCRYPT_CHAIN_MODE_CBC);
-        private static readonly SafeAlgorithmHandle s_hAlgEcb = OpenRC2Algorithm(Cng.BCRYPT_CHAIN_MODE_ECB);
-
-        internal static SafeAlgorithmHandle GetSharedHandle(CipherMode cipherMode)
+        internal static SafeAlgorithmHandle GetHandle(CipherMode cipherMode, int effectiveKeyLength)
         {
-            // Windows 8 added support to set the CipherMode value on a key,
-            // but Windows 7 requires that it be set on the algorithm before key creation.
+            // Windows 8 added support to set CipherMode and EffectiveKeyLength on a key,
+            // but Windows 7 requires that they be set on the algorithm before key creation.
+            // Unlike the other SymmetricAlgorithm types that cache the algorithm based on CipherMode,
+            // RC2 creates a new algorithm each time since it must set EffectiveKeyLength.
             switch (cipherMode)
             {
                 case CipherMode.CBC:
-                    return s_hAlgCbc;
+                    return OpenRC2Algorithm(Cng.BCRYPT_CHAIN_MODE_CBC, effectiveKeyLength);
```

Comment:
I was going to ask if it made sense to use a cached handle for default effective keylength; but then I realized that a) that would make the code more complicated and b) this class should have nearly 0 usage so it doesn't matter.